### PR TITLE
prepare changelog for v0.17.0

### PR DESCRIPTION
**Pull request by kpcraig**

### CODE REVIEW

**Comment by kpcraig** [comment]
security scan run manually:
```
≻ make dev && scan binary bin/vault-plugin-secrets-azure
==> Checking that code complies with gofmt requirements...
go generate
==> Removing old directory...
==> Building...

==> Results:
total 300992
-rwxr-xr-x  1 kay.craig  staff   147M Feb  1 16:07 vault-plugin-secrets-azure
✓ Scanned file:{path:"bin/vault-plugin-secrets-azure"} in 6.4s - no results found
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,16 @@
-## Unreleased
+## v0.16.4
```
Comment (benashz) [inline CHANGELOG.md:1]: Should this be `v0.17.0`?

Comment (kpcraig) [reply CHANGELOG.md:1]: yep!

**Review by benashz — APPROVED** [review]
LGTM!

**Pull request merged**